warehouse/query-core/src/main/java/datawave/query/tables/keyword/StatefulKeywordUUIDChainStrategy.java

```diff
@@ -50,13 +63,20 @@ public void setBatchSize(int batchSize) {
     protected Query buildLatterQuery(Query initialQuery, Iterator<Entry<Key,Value>> initialQueryResults, String latterLogicName) {
         log.debug("buildLatterQuery() called...");
 
+        log.debug("initializing extractors");
+        for (TagCloudInputExtractor tagCloudInputExtractor : extractors) {
+            tagCloudInputExtractor.initialize(initialQuery);
+        }
+
+        log.debug("building query and extracting data");
         String queryString = captureResultsAndBuildQuery(initialQueryResults, batchSize);
 
         if (log.isDebugEnabled()) {
             log.debug("latter query is " + queryString);
         }
 
-        if (queryString == null || queryString.isBlank()) {
+        // as long as there are extractors it is okay for an empty query string, if neither there is nothing to do
+        if (extractors.isEmpty() && (queryString == null || queryString.isBlank())) {
```

Comment:
This is one case where I like StringUtils - to simplify the conditional...
```suggestion
        if (extractors.isEmpty() && StringUtils.isBlank(queryString)) {
```